app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,17 @@
+class CartsController < ApplicationController
+  include ActionView::Helpers::TextHelper
+
+  def create
+    item = Item.find(params[:item_id])
+
+    @cart.add_item(item.id)
+    session[:cart] = @cart.contents
+
+    flash[:notice] = "You now have #{pluralize(@cart.count_of(item.id), item.name)}."
+    redirect_back(fallback_location: items_path)
```

Comment:
New feature! I was using redirect_to :back, but got a nice warning saying it's deprecated for the next version